### ISSUE TITLE
GradientTextures uses sRGB colorspace in HTML only

### ISSUE THREAD

**Issue by robpc** [godotengine/godot#48460]
**Similar Issues**
* https://github.com/godotengine/godot/issues/25142 - Provides a clue as to what is happening, but not related to the html vs editor inconsistency
* https://github.com/godotengine/godot/issues/48039 - Mine is more of an inconsistency in platforms and this is a larger philosophical discussion

**Godot version:**
3.3.stable

**OS/device including version:**

* Card Model: RTX 2080
* Driver: GeForce 466.27
* OS: Windows 10
* Backend: GLES3

**Issue description:**
When using a Gradient Texture as a `sample2d` uniform in a shader, the colorspace appears to not match the editor or desktop exports when exporting to HTML. When searching for this issue I found the discussion in https://github.com/godotengine/godot/issues/25142 which was helpful but while the solution "worked" in html it did not work in the editor or the windows export. On further examination, it seems like maybe the flag `OUTPUT_IS_SRGB` should be set in html but isn't.

Here is a screenshot of the sample project. 
<img width="626" alt="shader-color-in-html-test-full" src="https://user-images.githubusercontent.com/3671561/117045171-3745c000-acdd-11eb-88af-283ab7036206.png">

*The left/right boxes are SpatialMaterials with the albedo values from both ends of the two point gradient used in the center box.
* The center box contains three variations on the shader in this order:
  * Original sampling from the GradientTexture directly
  * Sampling from the GradientTexture converting using the solution from https://github.com/godotengine/godot/issues/25142 if the flag is not set
  * Using two color values to create a gradient manually
  * The last line shows if the flag is set or not
```glsl
    vec3 color = vec3(0.0);
    if (OUTPUT_IS_SRGB) {
        color = vec3(1.0);
    }
    ALBEDO = color ;
```

**Steps to reproduce:**
* Create a mesh and assign it a shader material
* Add the shader code 
```glsl
shader_type spatial;
render_mode cull_disabled, unshaded;

uniform sampler2D color_ramp : hint_black_albedo;

void fragment() {
    ALBEDO = texture(color_ramp, vec2(1.0 - UV.y, 0)).rgb ;
}
```
* Add a GradientTexture to the uniform, I used the color values `134f00` and `35dd00`.
* Export to both windows and html and compare

**Minimal reproduction project:**
https://github.com/robpc/godot-shader-color-in-html-test

**Comment by Calinou** [comment]
In my experience, Chrome and Firefox will often have different color output on the same machine. I'm not sure if we should attempt to correct this.

**Comment by robpc** [comment]
I don't think it is a browser issue since they are all consistent in the output.
<img width="1439" alt="shader-color-in-html-test-browsers" src="https://user-images.githubusercontent.com/3671561/117068350-2dca5100-acf9-11eb-8561-c7c56fca1784.png">

But I understand if GLES3 HTML bugs are not on the roadmap. I will go with the manual gradient for my needs just because the project this came up in runs at 1fps in GLES2 :P
